flink-yarn/src/main/java/org/apache/flink/yarn/configuration/YarnConfigOptions.java

```diff
@@ -52,7 +52,74 @@
 		key("yarn.per-job-cluster.include-user-jar")
 			.defaultValue("ORDER");
 
+	/**
+	 * The vcores exposed by YARN.
+	 */
+	public static final ConfigOption<Integer> YARN_VCORES =
+		key("yarn.containers.vcores")
+		.defaultValue(Integer.MAX_VALUE);
```

Comment:
Personally i think we should have methods that allow overriding the default for non-String options; I've already opened FLINK-6638 for this.

Let's use -1 as the default instead, it's a more "expected" invalid value i suppose.